library/src/main/java/com/whatsapp/stringpacks/StringPacks.java

```diff
@@ -192,7 +192,8 @@ private static Locale getLocaleFromContext(@NonNull Resources resources) {
    */
   private static List<String> getParentLocales(@NonNull Locale locale) {
     final ArrayList<String> parents = new ArrayList<>();
-    parents.add(locale.getLanguage());
+    parents.add(String.format("%s" + (!locale.getCountry().equals("") ? "-%s" : ""),
```

Comment:
`getParentLocales` method provides locale hierarchy in which StringPacks tries to find a string. Hence, it also plays an important role in fallback mechanism.
Having said that, there is a possibility that people can pack strings for `es` and `es-MX` together in `strings_es.pack` and want a fallback mechanism to be `es-MX` -> `es` -> `en` (from Android's resources).

This change won't be able to enable this fallback logic from `es-MX` to `es` and fails the purpose of packing similar strings together via `pack_id_mapping` property of `config.json`.

In my opinion we must open up an API for apps to provide their own fallback parent locales based on their config or preference.

```suggestion
    if (stringPackLocaleManager == null || stringPackLocaleManager.shouldAddLanguageAsParentForLocale(locale)) {
        parents.add(locale.getLanguage());
    }
    if (stringPackLocaleManager != null) {
        parents.add(stringPackLocaleManager.getParentLocaleForLocale(locale));
    }
```

Hence, we can expose an interface `StringPackLocaleManager` in `StringPacks.java` and a getter/setter method for the implementation of this interface. The apps can create an implementation of this interface and set their implementation from Application class.